go-backend/service/session_http.go

```diff
@@ -0,0 +1,75 @@
+package service
+
+import (
+	"errors"
+	"joshsoftware/peerly/config"
+	"joshsoftware/peerly/db"
+	"net/http"
+	"time"
+
+	"github.com/dgrijalva/jwt-go"
+	logger "github.com/sirupsen/logrus"
+)
+
+type Claims struct {
+	UserID         int64     `json:"user_id"`
+	ExpirationDate time.Time `json:"expiration_date"`
+	jwt.StandardClaims
+}
+
+var errInvalidToken = errors.New("Invalid Token")
+var errMissingAuthHeader = errors.New("Missing Auth header")
+
+func getClaims(token string) (claims *Claims, err error) {
+	// Initialize a new instance of `Claims`
+	claims = &Claims{}
+
+	var newTkn *jwt.Token
+	newTkn, err = jwt.ParseWithClaims(token, claims, func(token *jwt.Token) (interface{}, error) {
+		return config.JwtKey(), nil
+	})
+
+	if !newTkn.Valid {
+		err = errInvalidToken
+		return
+	}
+
+	return
+}
+
+func handleLogout(deps Dependencies) http.HandlerFunc {
+	return http.HandlerFunc(func(rw http.ResponseWriter, req *http.Request) {
+		token := req.Header.Get(authHeader)
+		if token == "" {
+			rw.WriteHeader(http.StatusBadRequest)
+			logger.WithField("err", errMissingAuthHeader.Error()).Error("Auth Header missing")
+		}
+
+		claims, err := getClaims(token)
+		if err != nil {
+			if err == jwt.ErrSignatureInvalid {
+				rw.WriteHeader(http.StatusUnauthorized)
+				return
+			} else if err == errInvalidToken {
+				rw.WriteHeader(http.StatusUnauthorized)
+				return
+			}
```

Comment:
I recommend adding an additional `else` block here as a catch-all, with a `http.StatusUnauthorized` header being written out. Maybe we can later add some conditional logging (e.g. "if verbose mode, log user id and (maybe?) info about the token")

Reply:
Good Idea, I am thinking of using `rw.WriteHeader(http.StatusBadRequest)` for catch-all. as status `Bad Request` feels good fit here. Implemented in fb35e6c